pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -312,16 +325,36 @@ func TestResponseRoundtrip(t *testing.T) {
 			require.NoError(t, err)
 			assert.Equal(t, tc.expected, decoded)
 
+			metrics, err := util.NewMetricFamilyMapFromGatherer(reg)
+			require.NoError(t, err)
+			durationHistogram, err := findHistogramMatchingLabels(metrics, "cortex_frontend_query_response_codec_duration_seconds", "format", "json", "operation", "decode")
+			require.NoError(t, err)
+			require.Equal(t, uint64(1), *durationHistogram.SampleCount)
```

Comment:
(nit) Is it worth adding a check that the SampleSum is within an expected time duration? (even though that may change as encoding/decoding is improved)

Reply:
Can't hurt, it would catch any instances of getting the calculation wrong. Although we can't be too strict as that will make the test flaky.

Added in 0a6fe90.